src/runtime/virtcontainers/clh.go

```diff
@@ -1147,19 +1149,40 @@ func (clh *cloudHypervisor) addNet(e Endpoint) error {
 		return errors.New("net Pair to be added is nil, needed to get TAP path")
 	}
 
-	tapPath := netPair.TapInterface.TAPIface.Name
-	if tapPath == "" {
-		return errors.New("TAP path in network pair is empty")
+	fds := make([]int32, 0)
+	for _, f := range netPair.TapInterface.VMFds {
+		fd := f.Fd()
+		flags, err := unix.FcntlInt(fd, unix.F_GETFD, 0)
+		if err != nil {
+			return err
+		}
+
+		// Let's remove the FD_CLOEXEC flag from the file descriptor,
+		// so it doesn't get closed when executing cloud-hypervisor.
+		//
+		// This is a workaround discussed with the cloud-hypervisor
+		// team, and we should keep it while they can't support file
+		// descriptors being passed down using syscall.UnixRights() /
+		// unix.UnixRights(), as done for QEMU.
+		flags &^= unix.FD_CLOEXEC
```

Comment:
It would be good to add an explicit test for this: set the `FD_CLOEXEC` flag and check to ensure it gets removed.